compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/DecodeV1.hs

```diff
@@ -84,39 +99,41 @@ decodeName
     :: Util.EitherLike TL.Text Int32 e
     => (T.Text -> a) -> Maybe e -> Decode a
 decodeName wrapName mbStrOrId = mayDecode "name" mbStrOrId $ \strOrId -> do
-    mangled <- case Util.toEither strOrId of
-        Left str -> pure $ decodeString str
+    unmangledOrErr <- case Util.toEither strOrId of
+        Left str -> pure $ decodeMangledString str
         Right strId -> lookupString strId
-    decodeNameString wrapName mangled
+    decodeNameString wrapName unmangledOrErr
 
-decodeNameString :: (T.Text -> a) -> T.Text -> Decode a
-decodeNameString wrapName mangled =
-    case unmangleIdentifier mangled of
+decodeNameString :: (T.Text -> a) -> (T.Text, Either String UnmangledIdentifier) -> Decode a
+decodeNameString wrapName (mangled, unmangledOrErr) =
```

Comment:
My impression is that we only need `mangled` here for the error message. Is that correct? If so, could we probably make the error message returned by `unmangleIdentifier` better and remove the additional complexity here and in the surroundings?

Reply:
Good point, I’ve shuffled it around.